lib/dead_end/who_dis_syntax_error.rb

```diff
@@ -53,9 +53,11 @@ def on_parse_error(msg)
       when /expecting end-of-input/
         @error_symbol = :unmatched_syntax
         @unmatched_symbol = :end
-      when /unexpected `end'/,      # Ruby 2.7 and 3.0
-           /unexpected end/,        # Ruby 2.6
-           /unexpected keyword_end/i # Ruby 2.5
+      when /unexpected `end'/,          # Ruby 2.7 and 3.0
```

Comment:
What do you think about adding a case statement above the `unexpected end` case. Something like:

```ruby
when /unexpected .* expecting '(?<unmatched_symbol>.*)/
  match = $LAST_MATCH_INFO
  @unmatched_symbol = match[:unmatched_symbol].to_sym
  @error_symbol = :unmatched_syntax
```

I'm wondering if that will allow us to simplify this a bit. My worry is the larger this case becomes the more likely accidentally firing this logic on an unrelated parse error becomes larger. By moving the "expecting <symbol>" logic above this case in the detection, we can avoid trying to catch some of these seemingly unrelated other exception types.

Reply:
@schneems thanks, updated! Let me know what you think. 

I checked and the only one that doesn't match that regexp now is `unexpected `end'` so I left that one there, and the rest are covered by that new regexp now :+1: Also had to use `$1` instead, because for some reason $LAST_MATCH_INFO comes back as `nil` 🤔

Reply:
So I had to also leave the cases for 2.5 and 2.6, although we could use a regex there too.. what do you think?

Reply:
This is great, exactly what I had in mind